### ISSUE TITLE
Alt text added to Image Settings in Image block is not retained for reuse

### ISSUE THREAD

**Issue by abrightclearweb** [WordPress/gutenberg#8941]
**Describe the bug**
Uploading an image direct to a Gutenberg post puts the image directly in the post, bypassing the Media Library dialog with Attachment Details.

If you select an image to add from the Media Library, you see the Select or Upload Media screen where the Attachment Details show when you have selected something. There you can add caption, alt text and description which are retained for subsequent usage. If you add alt text here it's copied over to the Alt text box in Image Settings.

For a brand new upload a user won't see the Media Library Attachment Details, so it's logical to add alt text in the Image Settings panel. In that case, it's not saved for reuse.

**To Reproduce**
1. Upload an image to a new post. 
2. Complete Alt text in Image Settings.
3. Save the post.
4. Create a new post.
5. Add the same image to the post.
6. No Alt text is shown.

**Expected behavior**
I expected that the alt text would be saved for that image for reuse. If I add alt text when uploading an image in the classic editor, the alt text is retained for the next time I use the image.

I could go to the Media Library and edit the image there to add alt text, but that breaks my workflow.

**Screenshots**

Image uploaded, alt text typed in Image Settings
![image uploaded and alt text added](https://user-images.githubusercontent.com/7430856/44054201-dff0e34c-9f39-11e8-8678-2bb4b948f5b3.png)

Alt text shows in the post
![alt text on the front end](https://user-images.githubusercontent.com/7430856/44054227-f0f51ff0-9f39-11e8-8a2c-ebee8f991827.png)

Image selected for reuse
![image reselected - no alt text](https://user-images.githubusercontent.com/7430856/44054260-055f9f92-9f3a-11e8-8bc6-e9ccb79489bc.png)

No alt text :(
![alt text box is empty](https://user-images.githubusercontent.com/7430856/44054274-0ef49ba2-9f3a-11e8-9afe-fa73584b78a3.png)

**Desktop (please complete the following information):**

   - OS: Windows 10 64-bit
   - Browser: Firefox Quantum
   - Version 61.0.1 (64-bit)

**Additional context**
- Gutenberg 3.5.0

**Comment by slattery** [comment]
This is true for Alt and Caption in v3.7.0 as well.    I would also want to update the Description field here.

**Desktop (please complete the following information):**

OS: OS X High Sierra
Browser: Chrome 
Version 68.0.3440.106 (64-bit)

**Additional context**

Gutenberg 3.7.0

**Comment by slattery** [comment]
This is still true for Alt and Caption in 4.3

**Comment by mrwweb** [comment]
This is still true as of Gutenberg 5.1 and WordPress 5.1. It makes it significantly harder to maintain an accessible and useful media library.

#11054 is related in that many of the new image upload flows either don't allow editing image meta or don't sync that meta back to the Media Library. This is going to lead to significantly bloated and unsearchable media libraries.

**Comment by TheMangoTrain** [comment]
Still an issue Nov 9, 2019
Gutenberg plugin 6.8.0
WP 5.2.4

**Comment by enriquesanchez** [comment]
There is a similar Core Trac ticket:

[Alt Text Not Saving Outside of Library](https://core.trac.wordpress.org/ticket/49165)

The conclusion from the WP Accessibility team was that this works as expected:

>This ticket was discussed during today's accessibility bug-scrub. This works as expected but it's clear that the user interface could be more... clear. There's a ticket for that! See #47456.

From https://core.trac.wordpress.org/ticket/47456:

> A common misconception is that the image alt text should always be a "description of the image". In most of the cases, this is misleading. Actually, the image alt text needs to describe the purpose of the image in its specific context. For more details, see the W3C Alt Text Decision Tree tutorial (https://www.w3.org/WAI/tutorials/images/decision-tree/).

>WordPress stores a "default" alt text in the media object. While storing a default value may help users when they build their content, it also promotes a misunderstanding of the purpose of the alt text.

>In the accessibility team, we think this is more an user interface problem rather than a data model problem. The user interface should ensure users clearly understand that alt attributes are context sensitive and that the "default" alt value needs to be changed (or even removed) based on the specific usage.

---

Please feel free to reopen this issue if the discussion needs to continue.

**Comment by vigilantspb** [comment]
WP.. surprises me. Everyone wants to have reusable alt, but can go and f yourself - it's expected)
It used to work in old versions, now it's not and it's ok? It was useful for searching old uploaded media at least...

**Comment by kbarkhurst** [comment]
Updating my original comment: I agree with the WP Core's reply and the user's complaints. I spent some time troubleshooting this and as an update, I did discover a workaround that allows for a single upload from a post or page that adds the alternative description to both the Library and Post. 

When you click to Upload an image, choose the text link that reads "Media Library" instead of using the blue Upload button. Then click the first Tab that reads Upload. When you upload this way, you're uploading directly to the Media Library and can add a Media Library description during the upload process. Then click Select. Your image is added to your post and media library with the same description. If you choose, you can now change your alternative text for just this post's image using the block area. 

This is a nice feature. It encourages what the Core team wanted: having users make post-specific alts because most users will use the blue Upload button. It also gives users who want the shared functionality an option to do that, too. Thank you!

**Comment by samuelmente** [comment]
Hey everyone,

I just ran into this while working on a site and was confused why alt texts set in the media library weren’t showing on the frontend.

I get that alt text should be context-specific. However, in cases where no alt text is set in Gutenberg at all, wouldn’t it be better to fall back to the media library value instead of outputting an empty alt attribute?

A possible approach:

* If the alt text in Gutenberg is empty → use the media library value
* If an alt text is explicitly set in Gutenberg → leave it unchanged

This would improve accessibility in cases where alt text is unintentionally missing, without overriding deliberate choices.
